src/main/java/de/zalando/aruha/nakadi/controller/SubscriptionController.java

```diff
@@ -0,0 +1,97 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.UUID;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+import static org.zalando.problem.spring.web.advice.Responses.create;
+
+@RestController
+@RequestMapping(value = "/subscriptions")
+public class SubscriptionController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SubscriptionController.class);
+
+    private final SubscriptionDbRepository subscriptionRepository;
+
+    private final EventTypeRepository eventTypeRepository;
+
+    @Autowired
+    public SubscriptionController(final SubscriptionDbRepository subscriptionRepository,
+                                  final EventTypeRepository eventTypeRepository) {
+        this.subscriptionRepository = subscriptionRepository;
+        this.eventTypeRepository = eventTypeRepository;
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<?> createOrGetSubscription(@Valid @RequestBody final Subscription subscription,
+                                                     final Errors errors, final NativeWebRequest nativeWebRequest) {
+        if (errors.hasErrors()) {
+            return create(new ValidationProblem(errors), nativeWebRequest);
+        }
+
+        try {
+            // check that event types exist
+            final List<String> noneExistingEventTypes = newArrayList();
+            for (final String etName : subscription.getEventTypes()) {
+                try {
+                    eventTypeRepository.findByName(etName);
+                } catch (NoSuchEventTypeException e) {
+                    noneExistingEventTypes.add(etName);
+                }
+            }
+            if (!noneExistingEventTypes.isEmpty()) {
+                final String errorMessage = "Failed to create subscription, event type(s) not found: '" +
+                        StringUtils.join(noneExistingEventTypes, "','") + "'";
+                LOG.debug(errorMessage);
+                return create(NOT_FOUND, errorMessage, nativeWebRequest);
+            }
+
+            // generate subscription id and try to create subscription in DB
+            subscription.setId(UUID.randomUUID().toString());
+            subscriptionRepository.saveSubscription(subscription);
```

Comment:
If client will provide created_at field, than incorrect data will be written. I think, that we need two objects - one to store data and the second one to return it. And the second one is a child of the first one.